### PR TITLE
Align Sync API's with Cocoa

### PR DESCRIPTION
**Pull request by cmelchior**
Align Sync API's with Cocoa.

Reasoning:
> “Log in” is a phrasal verb, while “login” is a noun. Given that the method in question performs an action, the verb is the correct choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 ### Breaking Changes
 
+* [ObjectServer] Renamed `SyncUser.currentUser()` to `SyncUser.current()`.
+* [ObjectServer] Renamed `SyncUser.login(...)` and `SyncUser.loginAsync(...)` to `SyncUser.logIn(...)` and `SyncUser.logInAsync(...)`.
```
Comment (Zhuinden) [inline CHANGELOG.md]: what the :smile:

**Comment by nhachicha** [comment]
@cmelchior need to update the examples though

**Comment by cmelchior** [comment]
Doh, thanks

File: CHANGELOG.md (hunk 1 above)
Comment (Zhuinden) [reply CHANGELOG.md]: really though, are guys sure it's not cocoa who should remove the uppercase?

I don't think I've ever seen it as `logIn` anywhere.

**Comment by Zhuinden** [comment]
I've really never seen APIs call it `logIn`, but I guess it's your call ¯\\\_(ツ)_/¯

**Pull request merged**